lib/deref/index.js

```diff
@@ -51,6 +51,13 @@ module.exports = function deref(boundJSONArg) {
         nextPath[++nextPathLength] = reversedKeys[i];
     }
 
+    // This is the key for creating a path from key/parent combo's only
+    // and there was already a bound path.  If we follow a reference (path
+    // variable is defined) then we do not need to concat any paths.
+    if (!path && this._path.length) {
```

Comment:
We should add a test case for the reference case also [ the `if (path && this._path.length)` case ]. I can add that.